File: leetcode/question003/main.js
/**
 * @param {string} s
 * @return {number}
 */

var lengthOfLongestSubstring = function (s) {
    var result = 0;
    for (var i = 0; i < s.length; i++) {
        var temp = getLength(s, i);
        result = result > temp ? result : temp;
    }
    return result;
};

var getLength = function (s, index) {
    for (var i = index + 1; i < s.length; i++) {
        var str = s.substring(index, i);
        var c = s.charAt(i);
        if (str.indexOf(c) >= 0) {
            return i - index;
        } else {
            continue;
        }
    }
    return s.length - index;
};

console.log(lengthOfLongestSubstring("abcabcbb"));

File: leetcode/question003/main.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        def getLength(s, index):
            i = index+1
            while i < len(s):
                str = s[index:i]
                c = s[i]
                if c in str:
                    return i-index
                else:
                    i = i+1
                    continue
            return len(s)-index
        result = 0
        for index in range(len(s)):
            temp = getLength(s, index)
            result = result if result > temp else temp
        return result
